File: javae/test/IceE/location/TestI.java
// **********************************************************************
//
// Copyright (c) 2003-2008 ZeroC, Inc. All rights reserved.
//
// This copy of Ice-E is licensed to you under the terms described in the
// ICEE_LICENSE file included in this distribution.
//
// **********************************************************************

import Test.*;

public class TestI extends _TestIntfDisp
{
    TestI(Ice.ObjectAdapter adapter1, Ice.ObjectAdapter adapter2, ServerLocatorRegistry registry)
    {
        _adapter1 = adapter1;
        _adapter2 = adapter2;
        _registry = registry;

        _registry.addObject(_adapter1.add(new HelloI(), _adapter1.getCommunicator().stringToIdentity("hello")));
    }

    public void
    shutdown(Ice.Current current)
    {
        _adapter1.getCommunicator().shutdown();
    }

    public HelloPrx
    getHello(Ice.Current current)
    {
        return HelloPrxHelper.uncheckedCast(_adapter1.createIndirectProxy(_adapter1.getCommunicator().stringToIdentity("hello")));
    }

    public HelloPrx
    getReplicatedHello(Ice.Current current)
    {
        return HelloPrxHelper.uncheckedCast(_adapter1.createProxy(_adapter1.getCommunicator().stringToIdentity("hello")));
    }

    public void
    migrateHello(Ice.Current current)
    {
        final Ice.Identity id = _adapter1.getCommunicator().stringToIdentity("hello");
        try
        {
            _registry.addObject(_adapter2.add(_adapter1.remove(id), id));
        }
        catch(Ice.NotRegisteredException ex)
        {
            _registry.addObject(_adapter1.add(_adapter2.remove(id), id));
        }
    }

    private ServerLocatorRegistry _registry;
    private Ice.ObjectAdapter _adapter1;
    private Ice.ObjectAdapter _adapter2;
}

File: objc/test/Ice/facets/TestI.m
// **********************************************************************
//
// Copyright (c) 2003-2008 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

#import <Ice/Ice.h>
#import <TestI.h>

@implementation AI
-(NSString*) callA:(ICECurrent*)current
{
    return @"A";
}
@end

@implementation BI
-(NSString*) callA:(ICECurrent*)current
{
    return @"A";
}
-(NSString*) callB:(ICECurrent*)current
{
    return @"B";
}
@end

@implementation CI
-(NSString*) callA:(ICECurrent*)current
{
    return @"A";
}
-(NSString*) callC:(ICECurrent*)current
{
    return @"C";
}
@end

@implementation DI
-(NSString*) callA:(ICECurrent*)current
{
    return @"A";
}
-(NSString*) callB:(ICECurrent*)current
{
    return @"B";
}
-(NSString*) callC:(ICECurrent*)current
{
    return @"C";
}
-(NSString*) callD:(ICECurrent*)current
{
    return @"D";
}
@end

@implementation EI
-(NSString*) callE:(ICECurrent*)current
{
    return @"E";
}
@end

@implementation FI
-(NSString*) callE:(ICECurrent*)current
{
    return @"E";
}
-(NSString*) callF:(ICECurrent*)current
{
    return @"F";
}
@end

@implementation GI
-(void) shutdown:(ICECurrent*)current
{
    [[[current adapter] getCommunicator] shutdown];
}
-(NSString*) callG:(ICECurrent*)current
{
    return @"G";
}
@end

@implementation HI
-(void) shutdown:(ICECurrent*)current
{
    [[[current adapter] getCommunicator] shutdown];
}
-(NSString*) callG:(ICECurrent*)current
{
    return @"G";
}
-(NSString*) callH:(ICECurrent*)current
{
    return @"H";
}
@end
